[16.0][FIX] cetmix_tower_yaml-export_files_issue

Mohajiro opened the pull request:
Add "file_id" in cx_tower_server_log and tests unitaires

Task: 4782

Override `_prepare_record_for_yaml` to set `code=False` when `source=='server'`
Add tests in `test_tower_yaml_mixin.py`

Task: 4783

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Enhanced YAML export/import to include file reference handling for server logs.
  - Added conditional YAML preparation for file records, clearing code content for server-sourced files.
- **Tests**
  - Added tests validating YAML export/import of server logs with file references and missing files.
  - Added tests verifying YAML preparation behavior based on file source type.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Mohajiro committed:
[FIX] cetmix_tower_yaml: export files issue

add "file_id" in cx_tower_server_log and tests unitaires

Task: 4782

ivs-cetmix commented on cetmix_tower_yaml/tests/test_server_log_file_id.py:
```diff
@@ -0,0 +1,95 @@
+    * import must restore the relation.
+    """
+
+    def setUp(self):
```
This is Odoo 16.0 Please user setUpClass instead

Mohajiro replied:
Ready! Thanks!

ivs-cetmix commented on cetmix_tower_yaml/tests/test_server_log_file_id.py:
You should not create a file just to test a single field. So either include this test in the existing file, or create a file per model.

Mohajiro replied:
I created one file for test for our model cx_tower_server_log.py so normally it's done!

ivs-cetmix reviewed (approved):
Code review LGTM

Mohajiro committed:
[IMP] cetmix_tower_yaml: file export improvements

Override `_prepare_record_for_yaml` to set `code=False` when `source=='server'`
Add tests in `test_tower_yaml_mixin.py`

Task: 4783

ivs-cetmix reviewed (approved):
Code review LGTM

Aldeigja reviewed (approved):
Functional LGTM

ivs-cetmix commented:
/ocabot merge patch

CetmixGitDrone commented:
Hey, thanks for contributing! Proceeding to merge this for you.
Prepared branch [16.0-ocabot-merge-pr-320-by-ivs-cetmix-bump-patch](https://github.com/cetmix/cetmix-tower/commits/16.0-ocabot-merge-pr-320-by-ivs-cetmix-bump-patch), awaiting test results.

CetmixGitDrone commented:
Congratulations, your PR was merged at fe3c0c268bbeef8e9768f03c48582d07501da424. Thanks a lot for contributing to cetmix. ❤️
